monai/run_inference_single_image.py

```diff
@@ -0,0 +1,280 @@
+"""
+Script to run inference on a MONAI-based model for contrast-agnostic soft segmentation of the spinal cord.
```

Comment:
i’ve started to review run_inference.py and then realized that this file also existed— and it has the same description— it creates a lot of duplication which is heavy to maintain— i’m sure there is an elegant way to combine both CLI scripts— @joshuacwnewton @mguaypaq @kousu might have ideas

Reply:
oh yeah this is an issue right now. In the last SCT meeting, we decided to stick to inference with single images to better fit into the `sct_run_batch` workflow. The file `run_inference.py` will be eventually removed once the `run_inference_single_image.py` is integrated into SCT.

Reply:
I very much agree that we should only be maintaining a single script here.

I tried checking the difference between the two scripts ([link](https://www.diffchecker.com/rk1dyc8M/)). I would expect only a little bit of difference (just related to iterating through the dataset vs. running a single image, etc.). However, the two scripts have diverged quite a lot -- I would guess that @naga-karthik has edited one of the scripts, but hasn't copied over the changes to the other script, so that one of the scripts has fallen out of date. (This is a good demonstration of @jcohenadad's point regarding maintenance -- it is very hard to keep two parallel scripts in sync.)

So, I would suggest to follow these steps:

1. First, try to minimize the difference between the two scripts. Go through the diff, then make sure both scripts are as similar as possible, avoiding inconsequential diffs where possible, e.g.:
   ![image](https://github.com/sct-pipeline/contrast-agnostic-softseg-spinalcord/assets/16181459/c97b7ba3-c6c9-4da1-b6c6-04a724aef4d2)
   The goal should be to limit the diffs to only the lines of code that are strictly relevant for single image vs. dataset processing.
2. Then, when you are left with only a small amount of differences, combine the two scripts. 
   For now, you can probably just stick to something simple like making a [mutually exclusive argument group](https://docs.python.org/3/library/argparse.html#mutual-exclusion) for `--path-json` and `--path-img`, so that only one can be specified. Then, having conditionals that will process the input differently depending on whether the script is in "dataset mode" or "image mode".

Reply:
Oh oops! I didn't see @naga-karthik's reply before I responded. I was a bit too slow.

If `run_inference.py` is obsolete, then it might be best to remove it now to avoid confusion? Unless there is a good reason to keep it around.

Reply:
Thanks @joshuacwnewton for the diff! I didn't realize that it has diverged that much 😅

I had to let `run_inference_single_image.py` diverge because I wanted to create a standalone script for inference. The `run_inference.py` depends on other functions and files which weren't necessary for single images. 

I see no point in maintaining the script for dataset processing since `sct_run_batch` will take care of it.

Reply:
ok great— do you want to delete run_inference then so as to not create confusion during code review?